Client/src/Components/Profile/UnitManagementSection/UnitForm.tsx

```diff
@@ -0,0 +1,51 @@
+import { Box, Button, Divider, Grid, ThemeProvider } from '@material-ui/core'
+import { FastField, FieldArray } from 'formik'
+
+import { MuiTextFieldFormik } from '../../Forms/FormikFields'
+import React from 'react'
+import { UnitList } from './UnitList'
+import { classStyles } from '../../../appTheme'
+import { disabledTheme } from '../../Forms/ComponentUpdate'
+
+interface IProps {
+  id?: number
```

Comment:
Why would we want to make id optional here?

Reply:
It's mainly optional for when users initial create a dimension. Until they're sent to the backend, a created dimension doesn't have an id. If id isn't set, it lets you change the name of the dimension as shown in the mockup.